src/containers/FormikForm/FormikCompetencesContent.tsx

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2020-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import React, { Fragment, useState, useEffect } from 'react';
+import { injectT } from '@ndla/i18n';
+import { FormPill } from '@ndla/forms';
+import { FieldProps, FormikActions, FormikValues } from 'formik';
+import { fetchCompetences } from '../../modules/draft/draftApi';
+import { fetchCompetenceTitle } from '../../modules/grep/grepApi';
+import { AsyncDropdown } from '../../components/Dropdown';
+import { isCompetenceValid } from '../../util/articleUtil';
+import { TranslateType } from '../../interfaces';
+
+interface Props {
+  t: TranslateType;
+  articleCompetences: string[];
+  field: FieldProps<string[]>['field'];
+  form: {
+    setFieldTouched: FormikActions<FormikValues>['setFieldTouched'];
+  };
+}
+
+interface Competence {
+  code: string;
+  title: string | undefined;
+}
+
+const FormikCompetencesContent = ({
+  t,
+  articleCompetences = [],
+  field,
+  form,
+}: Props) => {
+  const convertCompetencesToObject = async (competences: string[]) => {
+    return Promise.all(
+      competences.map(async c => {
+        return {
+          code: c,
+          title: await fetchCompetenceTitle(c),
+        };
+      }),
+    );
+  };
+  const [competences, setCompetences] = useState<Competence[]>([]);
+
+  const searchForCompetences = async (inp: string) => {
+    const result = await fetchCompetences(inp);
+    result.results = await convertCompetencesToObject(result.results);
+    return result;
+  };
+
+  useEffect(() => {
+    (async () => {
+      const comp = await convertCompetencesToObject(articleCompetences);
+      setCompetences(comp);
+    })();
+  }, []);
+
+  const updateFormik = (formikField: Props['field'], newData: string[]) => {
+    formikField.onChange({
+      target: {
+        name: formikField.name,
+        value: newData || null,
+      },
+    });
+  };
+
+  const createNewCompetence = async (newCompetence: Competence | string) => {
+    const competence = (newCompetence as Competence).code
```

Comment:
Er dette beste måten å gjøre dette på? Synes det virker litt tungvindt, men eg veit ikkje. Fint med en funksjon i stedet for to.

Reply:
Nå bruker jeg **createNewCompetence** for både create og add competences. Den ene funksjonen sender string som parameter den andre sender faktiske competence, så jeg må ha en måte å skille de på. Går også an å gjøre det med optional chaining men da må man håndtere undefined

Reply:
Kan den ikke bare alltid ta inn en streng og alltid sende inn en streng bare?